realis.go

```diff
@@ -563,29 +554,23 @@ func (r *realisClient) GetInstanceIds(key *aurora.JobKey, states map[aurora.Sche
 func (r *realisClient) GetJobUpdateSummaries(jobUpdateQuery *aurora.JobUpdateQuery) (*aurora.Response, error) {
 	var resp *aurora.Response
 	var err error
-
 	defaultBackoff := r.config.backoff
-	duration := defaultBackoff.Duration
-	for i := 0; i < defaultBackoff.Steps; i++ {
-		if i != 0 {
-			adjusted := duration
-			if defaultBackoff.Jitter > 0.0 {
-				adjusted = Jitter(duration, defaultBackoff.Jitter)
+	xerr := ExponentialBackoff(*defaultBackoff, func() (bool, error) {
+		resp, err = r.readonlyClient.GetJobUpdateSummaries(jobUpdateQuery)
+		if err != nil {
+			err1 := r.ReestablishConn()
+			if err1 != nil {
+				return false, nil
 			}
-			fmt.Println(" sleeping for: ", adjusted)
-			time.Sleep(adjusted)
-			duration = time.Duration(float64(duration) * defaultBackoff.Factor)
-		}
-		if resp, err = r.readonlyClient.GetJobUpdateSummaries(jobUpdateQuery); err == nil {
-			return response.ResponseCodeCheck(resp)
-		}
-		err1 := r.ReestablishConn()
-		if err1 != nil {
-			fmt.Println("error in ReestablishConn: ", err1)
 		}
+		return true, nil
+	})
 
+	if err != nil {
```

Comment:
should be xerr. Check on other APIs because the same is true for them.

Reply:
If there is an Xerr it definetely results in err.